api/org/commcare/api/persistence/SqlIndexedStorageUtility.java

```diff
@@ -0,0 +1,374 @@
+     */
+    public int getTotalSize() {
+        //serialize and test blah blah.
+        return 0;
+    }
+
+    /* (non-Javadoc)
+     * @see org.javarosa.core.services.storage.IStorageUtility#isEmpty()
+     */
+    public boolean isEmpty() {
+        return this.getNumRecords() > 0;
+    }
+
+
+    /* (non-Javadoc)
+     * @see org.javarosa.core.services.storage.IStorageUtility#readBytes(int)
+     */
+    public byte[] readBytes(int id) {
+        try {
+            Connection c = getConnection();
+            ResultSet rs = SqlHelper.selectForId(c, this.tableName, id);
+            byte[] caseBytes = rs.getBytes(org.commcare.modern.database.DatabaseHelper.DATA_COL);
+            c.close();
+            return caseBytes;
+        } catch (Exception e) {
+            System.out.println("SqlIndexedStorageUtility readBytes exception: " + e);
+        }
+        return null;
+    }
+
+    /* (non-Javadoc)
+     * @see org.javarosa.core.services.storage.IStorageUtility#remove(int)
+     */
+    public void remove(int id) {
+
+    }
+
+    /* (non-Javadoc)
+     * @see org.javarosa.core.services.storage.IStorageUtility#remove(org.javarosa.core.services.storage.Persistable)
+     */
+    public void remove(Persistable p) {
+        this.read(p.getID());
+    }
+
+    /* (non-Javadoc)
+     * @see org.javarosa.core.services.storage.IStorageUtility#removeAll()
+     */
+    public void removeAll() {
+
+    }
+
+    /* (non-Javadoc)
+     * @see org.javarosa.core.services.storage.IStorageUtility#removeAll(org.javarosa.core.services.storage.EntityFilter)
+     */
+    public Vector<Integer> removeAll(EntityFilter ef) {
+        return null;
+    }
+
+    /* (non-Javadoc)
+     * @see org.javarosa.core.services.storage.IStorageUtility#repack()
+     */
+    public void repack() {
+        //Unecessary!
+    }
+
+    /* (non-Javadoc)
+     * @see org.javarosa.core.services.storage.IStorageUtility#repair()
+     */
+    public void repair() {
+        //Unecessary!
+    }
+
+    /* (non-Javadoc)
+     * @see org.javarosa.core.services.storage.IStorageUtility#update(int, org.javarosa.core.util.externalizable.Externalizable)
+     */
+    public void update(int id, Persistable p) throws StorageFullException {
+        Connection c;
+        try {
+            c = getConnection();
+            SqlHelper.updateToTable(c, tableName, p, id);
+            c.close();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void setReadOnly() {
+        //TODO: This should have a clear contract.
+    }
+
+    private RuntimeException logAndWrap(Exception e, String message) {
+        RuntimeException re = new RuntimeException(message + " while inflating type " + prototype.getName());
+        re.initCause(e);
+        Logger.log("e", re.getMessage());
+        return re;
+    }
+
+    private final Vector<String> dynamicIndices = new Vector<String>();
```

Comment:
Content is added to `dynamicIndices`, but never read out again, is it useful for anything?